### PR TITLE
refactor(cli): simplify resume in-flight ownership

### LINKED ISSUE #7924: Simplify CLI resume-in-flight ownership
## Problem

`resumeInFlight` is canonically owned by `src/agent/runtime/resolveAndResumeStream.ts`, but the CLI relays its predicate through three one-consumer layers:

1. `ChatSessionController.isResumeInFlight`
2. `runChatTui` forwarding lambda
3. `CliAgentResumeHandler.isResumeInFlight`

The controller implementation simply re-exports the canonical runtime predicate. This creates duplicate dependency plumbing without alternate behavior or ownership.

## Acceptance criteria

- Keep `resumeInFlight` and `isResumeInFlight()` owned only by the shared runtime orchestrator.
- Keep the CLI handler registry only for the host-specific `tryResumeStream` callback.
- Preserve the current no-active-TUI-handler fallback (`false`).
- Remove controller/TUI predicate forwarding and mock-only assertions.
- Run focused resume/platform tests, typecheck, lint, format, and `git diff --check`.

## Scope

Behavior-preserving refactor only; no new state, abstraction, or fallback.

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

- remove `isResumeInFlight` from the chat controller and CLI handler registry
- read the canonical shared runtime predicate directly from the CLI platform adapter
- retain the active-handler guard so non-TUI CLI modes still report `false`
- replace mock-only predicate assertions with a real pending-resume lifecycle test

Closes #7924.

## Issue acceptance gates

- [x] `resumeInFlight` remains owned only by `resolveAndResumeStream.ts`
- [x] the CLI handler registry retains only host-specific `tryResumeStream` behavior
- [x] no active TUI handler still means no CLI resume is in flight
- [x] controller/TUI predicate forwarding is removed
- [x] an actual pending canonical resume verifies both handler-present and handler-absent behavior

## Single source of truth

`resumeInFlight` and `isResumeInFlight()` in the shared resume orchestrator own preparation state. The CLI adapter now consults that owner directly and uses handler presence only as its host-lifecycle gate.

## Duplication and abstraction budget

This removes two interface members, one forwarding lambda, one controller return field, and mock-only predicate plumbing. No state, fallback, wrapper, factory, schema, or dependency layer was added.

## Net elements (R6)

5 files changed; 0 files added/deleted; 40 insertions, 28 deletions, net +12 LoC. Production code is net -6 LoC; tests are net +18 LoC because the former predicate mock was replaced by a real pending-resume lifecycle oracle. No exported function, class, enum, or schema was added or removed.

## Consumer counts (R8)

- `ChatSessionController.isResumeInFlight`: 1 provider / 1 consumer before, 0 / 0 after.
- `CliAgentResumeHandler.isResumeInFlight`: 1 provider / 1 consumer before, 0 / 0 after.
- The canonical predicate keeps one direct CLI consumer, now in `agentResume.ts`.
- `isCliResumeInFlight` keeps one production consumer in the CLI `AgentResumePort` wiring.

## Host impact

Extension: no change.

Desktop: no change.

CLI: behavior preserved; active chat sessions report shared runtime preparation state, while headless/no-handler modes remain `false`.

## Scheduled refactoring

None.

## Validation

- final-head focused Vitest after review feedback: 3 files, 39 tests passed
- broader focused Vitest on the rebased production commit: 5 files, 55 tests passed
- full Vitest before the final non-overlapping rebase: 621 files, 5,455 tests passed; 1 file and 4 tests skipped
- final-head `npm run typecheck` passed, including all four desktop TypeScript configurations
- `npm run compile:fast` passed
- `npm run lint` passed with 0 errors and 52 pre-existing warnings; none are in changed files
- `npm run format` and pre-commit formatting hook passed
- `git diff --check` passed
- live `texra-local` PTY math delegation: one approved `review` child, 0 child tool calls, correct result `2√3`
- targeted PTY harness: 5 subagent follow-up, picker, viewer, and tab-focus scenarios passed

No changelog entry: this is behavior-preserving ownership cleanup.

---

> [!NOTE]
> **Low Risk**
> Behavior-preserving ownership cleanup with no new state layers; active-handler guard keeps headless CLI reporting false.
> 
> **Overview**
> **Resume preparation state** is no longer forwarded through the chat session controller or the CLI resume handler registry. `isCliResumeInFlight` now reads the canonical `isResumeInFlight()` from `resolveAndResumeStream`, gated only by whether an active TUI handler is registered (headless / no chat session still reports `false`).
> 
> The **handler surface shrinks** to host-specific `tryResumeStream` only: `ChatSessionController`, `CliAgentResumeHandler`, and `runChatTui` wiring drop the duplicate predicate and related mocks. Platform init still exposes `agentResume.isResumeInFlight` via `isCliResumeInFlight`.
> 
> Tests replace **mock-only predicate checks** with a lifecycle test that starts a real pending `resolveAndResumeStream` and asserts in-flight behavior with and without an active handler.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 417f7f22c06b8811d55e24a5754e963f8394c7d5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/test-kernel/cli/CliInitPlatform.vitest.mts [hunk 1]
```diff
@@ -215,21 +218,15 @@ describe('CLI platform init', () => {
...
     const dispose = setCliAgentResumeHandler({
       tryResumeStream,
-      isResumeInFlight,
     });
 
     try {
       await expect(
         nodePlatformOptions.agentResume.tryResumeStream('stream:cli-resume'),
       ).resolves.toBe(true);
       expect(tryResumeStream).toHaveBeenCalledWith('stream:cli-resume');
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/test-kernel/cli/CliInitPlatform.vitest.mts]: Addressed the concrete test fragility in `2d5d61d70`: the new `@agent/runtime/resolveAndResumeStream` mock is gone, and the host-agent mock ratchet now passes. I’m intentionally not adding branch tests for `activeHandler != null && isResumeInFlight(streamId)`: making the true branch controllable here would restore mock-only agent plumbing (or duplicate the full resume lifecycle), while handler registration is already covered through `tryResumeStream`, the canonical predicate has lifecycle coverage, and the active path was exercised in the live PTY delegation run. Given this PR’s test-cleanup goal, that extra assertion would cost more coupling than confidence.
